src/Compiler/Checking/AttributeChecking.fs

```diff
@@ -266,29 +266,28 @@ let langVersionPrefix = "--langversion:preview"
 
 /// Check F# attributes for 'ObsoleteAttribute', 'CompilerMessageAttribute' and 'ExperimentalAttribute',
 /// returning errors and warnings as data
-let CheckFSharpAttributes (g:TcGlobals) attribs m =
+let CheckFSharpAttributes (g:TcGlobals) attribs m = trackErrors {
     let isExperimentalAttributeDisabled (s:string) =
         if g.compilingFSharpCore then
             true
         else
             g.langVersion.IsPreviewEnabled && (s.IndexOf(langVersionPrefix, StringComparison.OrdinalIgnoreCase) >= 0)
 
-    if isNil attribs then CompleteD
+    if isNil attribs then do! CompleteD
     else
```

Comment:
Can the builder be moved to the `else` branch? That way we don't incur any overhead on the main path (no attributes).

Reply:
Yeah, I think so. Thanks for the review :)